qcodes/dataset/measurements.py

```diff
@@ -1055,6 +1060,17 @@ def add_subscriber(self: T,
 
         return self
 
+    def set_shapes(self, shapes: Shapes) -> None:
+        """
+        Set the shapes of the data to be recorded in this
+        measurement.
+
+        Args:
+            shapes: Dictionary from names of dependent parameters to a tuple
+                of integers describing the shape of the measurement.
+        """
+        self._shapes = shapes
```

Comment:
I pushed the validation into the dataset since that is kind of when you freeze the Interdeps and the shape. It seems like you can change them around as much as you like until then.

Reply:
> I pushed the validation into the dataset since that is kind of when you freeze the Interdeps and the shape.

that;'s fair, however Measurement implements registering parameters in a way that it always maintains a valid interdeps object, so following this consistency is it not fair to validate the shapes against current interdeps upon every set_shapes call as well?

Reply:
Perhaps, one could also argue that it is fine to set the shapes before registering the parameters but sure we can add that check

Reply:
Put in a validation here